bemuse/src/app/entities/Options.js

```diff
@@ -25,15 +25,15 @@ export const changeKeyMapping = (mode, key, keyCode) =>
 // Play mode
 export const playMode = (state) => state['player.P1.mode']
 export const changePlayMode = (mode) =>
-  u({
-    'player.P1.mode': mode,
-    'player.P1.panel': (panel) =>
-      panel === '3d' && mode !== 'KB' ? 'center' : panel,
+  produce((draft) => {
+    draft['player.P1'.mode] = mode
+    draft['player.P1.panel'] = (panel) =>
+      panel === '3d' && mode !== 'KB' ? 'center' : panel;
   })
 
 // Speed
 export const speed = (state) => state['player.P1.speed']
-export const changeSpeed = (speed) => u({ 'player.P1.speed': speed })
+export const changeSpeed = (speed) => produce((draft) => { draft['player.P1'.speed] = speed })
```

Comment:
That is a syntax error on this line.

Reply:
Is this should be written like `draft['player.P1.speed'] = speed` ?